### PR TITLE
doc: Add "Always Green CI" document

### LINKED ISSUE #417: Document the always green CI initiative
The "always green CI" initiative was initially proposed in https://docs.google.com/document/d/1vWCKYAQrxAFcuuxm60CBkqcO70SpmWoAqh_Hx4Gv5Jc/edit#heading=h.1hlxsk6xwa3i . Since then the initiative is operational and much have been discussed but we still lack of a centralized document to collect all information and policies agreed.

Let's have document in this repository or somewhere better (maybe on the community repo for broaden visibility) to centralize the knowledge.

### PR DESCRIPTION
**Pull request by wainersm**
Added a document for the always green CI initiative.

Fixes #417
Signed-off-by: Wainer dos Santos Moschetta <wainersm@redhat.com>

### CODE REVIEW

**Comment by wainersm** [comment]
Updated the document to pass the static-checks.

**Comment by wainersm** [comment]
@ariel-adam @chavafg @GabyCT @fidencio, here is the document for the "always green CI" initiative. I added some guidelines we didn't discuss on yesterday's meeting so feel free to send your comments.

Please review :)

File: Always_green_CI.md
Comment (ariel-adam) [inline Always_green_CI.md:7]: Change "section is discussed" to "section we discuss"

File: Always_green_CI.md
Comment (ariel-adam) [inline Always_green_CI.md:11]: Change "There was created" to "We have created"

**Review by ariel-adam — APPROVED** [review]

**Comment by ariel-adam** [comment]
@wainersm Added a few small comments however overall looks great :-)

File: Always_green_CI.md
Comment (jodh-intel) [reply Always_green_CI.md:11]: - s/run on daily/run on a daily/
- s/pinpoint/identify/
- s/we pursue to keep them "always green"/our aim is to ensure they always pass ("always green")/ maybe?

File: Always_green_CI.md
Comment (jodh-intel) [inline Always_green_CI.md:15]: Aside (random thoughts - nothing to do with this PR really ;): I do wonder if we should rename this site to `ci.katacontainers.io` (or even `status.katacontainers.io`) since:

- We might use something other than Jenkins one day.
- It's more discoverable / "guessable"

File: Always_green_CI.md
Comment (jodh-intel) [inline Always_green_CI.md:54]: I'd consider specifying another GitHub group here rather than individuals.

File: Always_green_CI.md
Comment (jodh-intel) [inline Always_green_CI.md:41]: Nit: Numbers less than ten should really be spelt as words.

File: Always_green_CI.md
Comment (jodh-intel) [inline Always_green_CI.md:13]: Can you link to a wiki page of owners here?

File: Always_green_CI.md
Comment (jodh-intel) [inline Always_green_CI.md:56]: Again, I'd consider creating a group (something like `kata-ci-admin` maybe?) with just Ariel in it rather than "hard-coding" his details here.

**Review by jodh-intel — COMMENTED** [review]
Thanks @wainersm. A few comments...

**Comment by jodh-intel** [comment]
One more comment: I wonder if it would be worth creating a diagram for the document to show the difference between the baseline CI and PR CI jobs? If you agree and don't plan to use ASCII art, see: https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#images.

**Comment by GabyCT** [comment]
thanks @wainersm it lgtm, just please do the changes that are required

**Review by GabyCT — APPROVED** [review]

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:7]: Done

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:11]: Done

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:3]: done

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:15]: IMHO it's a good idea, I've always avoided referring the technology  on names. I've seen the growth of a new phenomenon: projects written in rust being prefixed with 'rs'. Why??! I understand when the project is a re-write of a famous C library and so on; but for a brand new project it makes no sense :) 

`ci.katacontainers.io` looks good to me BTW.

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:41]: Interesting...never thought about it. Thx! and Done.

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:54]: I will change to `@kata-containers/ci`. Please let me know if I think it should be other.

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:13]: Later on this section I linked to the Jenkins page where the list of owners is. We don't have a wiki for it.

File: Always_green_CI.md
Comment (wainersm) [reply Always_green_CI.md:56]: Maybe create a group named like `ask-ariel-anything-about-your-future` :)

Serious now, have a group for dealing with administrative stuffs of this initiative looks good.

**Comment by wainersm** [comment]
> One more comment: I wonder if it would be worth creating a diagram for the document to show the difference between the baseline CI and PR CI jobs? If you agree and don't plan to use ASCII art, see: https://github.com/kata-containers/kata-containers/blob/main/docs/Documentation-Requirements.md#images.

It's a great idea; I really like charts and diagrams to express ideas however, in this case, I'm afraid I won't have free cycles to work on that anytime soon...

**Comment by wainersm** [comment]
Updated this document with changes to address (most but not all) the comments made by @ariel-adam and @jodh-intel.

**Comment by fidencio** [comment]
A head's up that may affect your PR.

In the Architecture Committee meeting from January 25th, 2022, the Architecture Committee has agreed on using the "Dismiss stale pull request approvals when new commits are pushed" configuration from GitHub.  It basically means that if your PR has been rebased or updated, the approvals given will be erased.

In order to minimize traumas due to the new approach, please, consider adding a note on the changes done before the rebase / force-push, and also pinging the reviewers for a subsequent round of reviews.

Thanks for your understanding!

Related issue: https://github.com/kata-containers/community/issues/249

File: README.md
Comment (jodh-intel) [reply README.md:57]: Maybe "baseline CI"?

File: Always_green_CI.md
Comment (jodh-intel) [reply Always_green_CI.md:17]: And as if by magic http://ci.katacontainers.io is now active and redirects to http://jenkins.katacontainers.io! :smile:

/cc @chavafg, @GabyCT, @kata-containers/ci.

**Review by jodh-intel — APPROVED** [review]
Wow, this has been open for waaaaay too long. Let's try to land this and iterate on it once it's in-tree...

**Pull request merged**
